encoding/ssh/filexfer/buffer.go

```diff
@@ -0,0 +1,268 @@
+package filexfer
+
+import (
+	"encoding/binary"
+	"errors"
+)
+
+// Various encoding errors.
+var (
+	ErrShortPacket = errors.New("packet too short")
+	ErrLongPacket  = errors.New("packet too long")
+)
+
+// Buffer wraps up the various encoding details of the SSH format.
+//
+// Data types are encoded as per section 4 from https://tools.ietf.org/html/draft-ietf-secsh-architecture-09#page-8
+type Buffer struct {
+	b   []byte
+	off int
+}
+
+// NewBuffer creates and initializes a new Buffer using buf as its initial contents.
```

Comment:
The argument is called b.

And the function doesn't just use it for initial contents. It takes ownership of b.

Reply:
Line was taken from `bytes.NewBuffer` https://golang.org/pkg/bytes/#NewBuffer

Reply:
Which says in the next sentence that the Buffer takes ownership :)

Reply:
👍 I shall import more of the godoc. 👍